src/com/google/enterprise/adaptor/SensitiveValueCodec.java

```diff
@@ -268,11 +271,27 @@ String getPrefix() {
    * -Dserver.keyAlias=adaptor \
    * -Dserver.secure=true
    * </pre>
+   *
+   * Example command line to run (passing the sensitive value on the command
+   * line):
+   * <pre>
+   * java \
+   * -Djavax.net.ssl.keyStore=keys.jks \
+   * -Djavax.net.ssl.keyStoreType=jks \
+   * -Djavax.net.ssl.keyStorePassword=changeit \
+   * -classpath 'adaptor-20130612-withlib.jar' \
+   * com.google.enterprise.adaptor.SensitiveValueCodec \
+   * -DsecurityLevel=ENCRYPTED \
+   * -Dserver.keyAlias=adaptor \
+   * -Dserver.secure=true
+   * 'sensitive value'
+   * </pre>
    */
   public static void main(String[] args) throws IOException {
     Config config = new Config();
     config.addKey("securityLevel", SecurityLevel.PLAIN_TEXT.toString());
     config.setValue("gsa.hostname", "not-used");
+    config.setValue("server.hostname", "not-used");
```

Comment:
In the class Config in line 210 the variable hostname is initialized with null. If an UnknownHostException is thrown a few lines below, this variable remains null. That being the case an InvalidConfigurationException will be thrown in line 894 when the config is validated ("validate" method) as step of the method "autoConfig" which is called in line 288 in SensitiveValueCodec. Hence I added this as config option with a dummy value (like "gsa.hostname").